### PR TITLE
Wheels up 🚀

### PR DESCRIPTION
**Pull request by talal**
/cc @reimannf

### CODE REVIEW

File: collectors/recon.go [hunk 1]
```diff
@@ -0,0 +1,637 @@
...
+			// most useful for alerting)
+			device := specialCharRx.ReplaceAllLiteralString(disk.Device, "")
+			ch <- prometheus.MustNewConstMetric(
+				clusterStorageUsedPercentByDiskDesc,
+				prometheus.GaugeValue, float64(disk.Used)/float64(disk.Size),
+				hostname, device,
+			)
+		}
+
+		usedPercent := float64(totalUsed) / float64(totalSize)
```
Comment (majewsky) [inline collectors/recon.go]: That's a fractional value, not a percentage value. Percentages are 0..100, but this ranges between 0 and 1.

File: collectors/recon.go [hunk 2]
```diff
@@ -0,0 +1,637 @@
...
+	)
+)
+
+var reconMD5Rx = regexp.MustCompile(
+	`(?m)^.* Checking ([\.a-zA-Z0-9_]+) md5sum(?:s)?\s*([0-9]+)/([0-9]+) hosts matched, ([0-9]+) error.*$`)
+
+func reconMD5Task(taskName, pathToReconExecutable string, ch chan<- prometheus.Metric) {
+	errCount := 0
+	cmdArgs := []string{"--md5"}
+	sendReconErrorCount(ch, cmdArgs, errCount)
```
Comment (majewsky) [inline collectors/recon.go]: When error paths are taken down below, `sendReconErrorCount()` will be called multiple times, which will cause multiple metrics with the same set of labels to be emitted. Prometheus probably won't like that. I suggest you change it like this:

```suggestion
	defer sendReconErrorCount(ch, cmdArgs, errCount)
```

and remove all the other `sendReconErrorCount()` invocations in this function. This causes the error count metric to be submitted when the function returns, using the then-current value of `errCount`.

(Same for all the other task functions.)

File: collectors/recon.go [hunk 3]
```diff
@@ -0,0 +1,624 @@
...
+				desc,
+				prometheus.GaugeValue, val,
+				hostname,
+			)
+		}
+		if gotErr {
+			errCount++
+		}
+
+		sendReconErrorCount(ch, cmdArgs, errCount)
```
Comment (majewsky) [inline collectors/recon.go]: This will not be called when the `return` is taken on line 314.

File: collectors/recon.go [hunk 4]
```diff
@@ -0,0 +1,624 @@
...
+				durDesc,
+				prometheus.GaugeValue, data.ReplicationTime,
+				hostname,
+			)
+		}
+		if gotErr {
+			errCount++
+		}
+
+		sendReconErrorCount(ch, cmdArgs, errCount)
```
Comment (majewsky) [inline collectors/recon.go]: Same problem as above.

File: collectors/recon.go (hunk 1 above)
Comment (majewsky) [reply collectors/recon.go]: 9c69349 does not fix this issue. The metric still has "percent" in its name and description.

File: collectors/recon.go (hunk 3 above)
Comment (talal) [reply collectors/recon.go]: There was also another problem with this approach. When `defer sendReconErrorCount()` was used at the top, the value of `errCount` was also evaluated and this meant that a later `errCount++` would have had no effect.

I have fixed this by wrapping these calls in an anonymous function.

While fixing it, I got an idea for a better approach for error handling. I will implement it along with the error test cases in a separate PR. For now, this should suffice.

File: collectors/recon.go (hunk 1 above)
Comment (majewsky) [reply collectors/recon.go]: This is resolved after discussion in Slack: The legacy implementation emits this metric in the same way with the same name, so we'll keep it like it is.

**Pull request merged**